pkg/controllers/managementuser/rbac/podsecuritypolicy/version.go

```diff
@@ -4,25 +4,33 @@ import (
 	"errors"
 	"fmt"
 
-	mVersion "github.com/mcuadros/go-version"
 	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	"k8s.io/apimachinery/pkg/util/version"
 )
 
-var errVersionIncompatible = errors.New("podsecuritypolicies are not available in Kubernetes v1.25 and above")
-var clusterVersionCheckErrorString = "failed to check cluster version compatibility for podsecuritypolicy controllers: %v"
+var ErrClusterVersionIncompatible = errors.New("podsecuritypolicies are not available in Kubernetes v1.25 and above")
 
-// checkClusterVersion tries to fetch a cluster by name, extract its Kubernetes version,
-// and check if the version is less than v1.25.
-func checkClusterVersion(clusterName string, clusterLister v3.ClusterLister) error {
+// CheckClusterVersion tries to fetch a cluster by name, extract its Kubernetes version,
+// and check if the cluster supports PodSecurityPolicies.
+func CheckClusterVersion(clusterName string, clusterLister v3.ClusterLister) error {
+	const mainErr = "failed to check cluster version compatibility for podsecuritypolicy controllers"
```

Comment:
I don't think you are getting much out of this variable, I would recommend forming specific error messages below and not relying on a "primary" error message.